[15.0][MIG] fieldservice: Migration to 15.0

victoralmau opened the pull request:
Superseed: https://github.com/OCA/field-service/pull/881

Migration to 15.0

Please @cesar-tecnativa and @CarlosRoca13 can you review it?

@Tecnativa TT35713

osi-scampbell committed:
[IMP] Various

max3903 committed:
[IMP] UI

max3903 committed:
[FIX] Fixes and improvements

max3903 committed:
[FIX] Wizard

Commit:
[FIX]Issue 1: Improper XML Domain for is_company 2. onchange territory_id filling proper manager details while creating FS location otherwise it makes null to respective region,branch,district

Commit:
[FIX]Issue: wrong widget many2many_tags for char field

Commit:
[FIX]auto fill territory preferred person on FSM Location

max3903 committed:
[IMP] Add duration and colors in views

max3903 committed:
[FIX] fieldservice

murtuzasaleh committed:
[MIG][WIP][12.0] fieldservice

murtuzasaleh committed:
[IMP] fieldservice: Added contacts in dependency.

murtuzasaleh committed:
[IMP] fieldservice: Improved action_view_contacts method.

murtuzasaleh committed:
[FIX][12.0] fieldservice

murtuzasaleh committed:
[12.0][IMP] fieldservice: Improved kanban view.

Commit:
[FIX]fieldservice migration issue related to less vs scss

osi-scampbell committed:
[FIX] #103 Search Worker Name

osi-scampbell committed:
[FIX] #104  Template Copy Notes Error

osi-scampbell committed:
[FIX] #73 Maximum Recursion on Location

osi-scampbell committed:
[IMP] #77 Search/Display Ref

osi-scampbell committed:
[IMP] #94 Show Assigned Button

osi-scampbell committed:
[IMP] #109 Stage Duplication Error

osi-scampbell committed:
[IMP] #107 Stage Type on List

osi-scampbell committed:
[IMP] #106 Require Stage Type

osi-scampbell committed:
[IMP] #44 Contact Location Context

osi-scampbell committed:
[IMP] #117 Mobile Field

osi-scampbell committed:
[IMP] #112 #105 Menu Changes

osi-scampbell committed:
[IMP] #97 Fedex Dist. Menu/Mobile Fix

osi-scampbell committed:
[IMP] #98 Move Instruction

osi-scampbell committed:
[IMP] #94 Office Hours on Location

osi-scampbell committed:
[IMP] #123 Group By Territory

osi-scampbell committed:
[IMP] #124 Operation Orders to 'All Orders'

osi-scampbell committed:
[IMP] #122 Past Due

osi-scampbell committed:
[IMP] #59 Substatus Visible NonDebug

osi-scampbell committed:
[IMP] #114 Working Schedules

osi-scampbell committed:
[IMP] #113 Advance/Previous Stage

osi-scampbell committed:
[IMP] Max Comments

osi-scampbell committed:
[IMP] Next -> Prev

osi-scampbell committed:
[IMP] Flake8 Errors

osi-scampbell committed:
[IMP] #129 Instruction Copying and Formatting

osi-scampbell committed:
flake8

osi-scampbell committed:
[IMP] fieldservice

murtuzasaleh committed:
[IMP][12.0] fieldservice: Improved onchange method.

max3903 committed:
[IMP] fieldservice: Add an option to auto-fill workers on location

max3903 committed:
[IMP] fieldservice: Add an option to auto-fill workers on location

max3903 committed:
[IMP] Sort workers on location

max3903 committed:
[FIX] Sort location by complete_name

max3903 committed:
[FIX] Show developer mode menu items to Manager group

max3903 committed:
[IMP] Hide Day Routes

max3903 committed:
[IMP] fieldservice: Add fieldservice_geoengine to settings

Commit:
[FIX] Issue 187: Expected Singleton

max3903 committed:
[IMP] fieldservice: Add setting for fieldservice_purchase

max3903 committed:
[FIX] fieldservice: AttributeError: 'bool' object has no attribute 'replace'

Commit:
[IMP]16335: fieldservice improvement to support preferred worker filter based on selected location #215

Commit:
[FIX]type

osi-scampbell committed:
Ticket #554 FSM Location Chatter

osi-scampbell committed:
[FIX] fieldservice (PM #515 and #577) (#230)

osi-scampbell committed:
[IMP] fieldservice: Add notes on location (PM #596) (#231)

osi-scampbell committed:
[IMP] #627 Filter Assigned like Request Workers

osi-scampbell committed:
[IMP] Location Report and Context Fix

[IMP] Remove Report

osi-scampbell committed:
[IMP] Add Equipment Tab

[IMP] Add Equipment Tab

[IMP] Remove field

osi-scampbell committed:
[IMP] Update Group

osi-scampbell committed:
[ADD] Calendar View FSM Orders

[IMP] Add fields

[ADD] Location Directions

Update fieldservice/models/fsm_order.py

Update fsm_order.py

[IMP] flake8

[IMP] Edit Copy Notes funcitonality

[IMP] Do Not Clear Equipment

[IMP] Edit Copy Notes funcitonality

osi-scampbell committed:
[IMP] Change duplicate label + Remove sequence

osi-scampbell committed:
[IMP] Delete Commented Sequence

osi-scampbell committed:
[IMP] Update FSM Order

[IMP] Check Before Stage Move

[IMP] Scheduled Check

[IMP] My Orders filter

kittiu committed:
[12.0][MIG] fieldservice_recurring

osi-scampbell committed:
[IMP] Remove Required for Repair/Maintenance

patrickrwilson committed:
[IMP] Add Today and Future Order Filters

Adds filters to show Today Orders and Future Orders.

brian10048 committed:
[IMP] fieldservice readme

kittiu committed:
[12.0][IMP] fieldservice, improve test

max3903 committed:
[FIX] fieldservice: Documentation syntax

osi-scampbell committed:
[IMP] fieldservice: Context Fix Active ID, Show Start with EnRoute, add Calendar to Teams (#260)

osi-scampbell committed:
[IMP] Bug #266 False ETA Error

[IMP] Bug #265 Linked Location Search Error

[IMP] Bug #272 Rate Not Declared

[IMP] Bug #263 equipment_count Wrong

[IMP] Bug #277 Worker Pricelists Computed Wrong

[IMP] Flake8 + Remove Debug

osi-scampbell committed:
[IMP] Move Location Down

hparfr committed:
Remove required attribute from view

hparfr committed:
bump numver

max3903 committed:
[FIX] fieldservice: Location name computation

max3903 committed:
[IMP] fieldservice: Report template

max3903 committed:
[IMP] fieldservice: Multi-company support

brian10048 committed:
[IMP] UI

minor UI improvements

hparfr committed:
Fix duration cacl on create

hparfr committed:
Update func naming

hparfr committed:
typo

patrickrwilson committed:
[UPD] Add to FS Settings

Added Project & Project Task module to the fieldservice settings.

patrickrwilson committed:
[UPD] Reviewer Suggestion

Removed setting groups as suggested by reviewer as they were not needed.

patrickrwilson committed:
[IMP] Add person_ids to search

Adding person_ids to allow searching by the requested workers.

osi-scampbell committed:
[IMP] Add stage_id to Locations List view

hparfr committed:
Remove most of the stages and its validation logic

hparfr committed:
Add name for inheritance

osi-scampbell committed:
[IMP] Add filters

[IMP] Request Late now Default not Computed

[IMP] Flake8

Commit:
[FIX]remove method call due to param error

hparfr committed:
Fix inheritance on default=

hparfr committed:
Same with fsm_team

hparfr committed:
Add name= on groups and page to simplify inheritance

max3903 committed:
[FIX] fieldservice

max3903 committed:
[FIX] fieldservice

patrickrwilson committed:
[IMP] Add CRM to Settings

Added CRM to field service settings section.

max3903 committed:
[FIX] fieldservice

max3903 committed:
[IMP] fieldservice_vehicle_stock

brian10048 committed:
[FIX] Unassigned Unscheduled Order Filters

brian10048 committed:
[FIX] Team kanban dashboard

Team kanban card was too narrow because SCSS was incorrect

patrickrwilson committed:
[IMP] fieldservice: Add Owned Locations to Partner (#376)

max3903 committed:
Added translation using Weblate (Spanish)

max3903 committed:
Added translation using Weblate (Spanish (Chile))

max3903 committed:
Added translation using Weblate (French)

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 1.7% (9 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 11.9% (63 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 39.2% (207 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 51.3% (271 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 64.4% (340 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 66.9% (353 of 528 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 99.1% (524 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 11.3% (60 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 13.8% (73 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 15.1% (80 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 15.5% (82 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/de/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 59.5% (315 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 67.9% (359 of 529 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

brian10048 committed:
[FIX] fieldservice

Users without FSM access could not open contacts

brian10048 committed:
[IMP] fieldservice

- Template form displayed the category field even though the group was disabled

brian10048 committed:
[IMP] fieldservice

- Added type to template
- Added test to fsm_order for template onchange method

osi-scampbell committed:
[IMP] Multi Company Stages & Teams

osi-scampbell committed:
[IMP] Flake8

osi-scampbell committed:
[IMP] Company Id == False

Commit:
Update fieldservice/models/fsm_order.py

Commit:
Update fieldservice/models/fsm_order.py

max3903 committed:
[FIX] fieldservice: PEP8 errors

hpatelserpentcs committed:
[IMP]Improved the calculation of scheduled dates and duration.

Commit:
Update fsm_order.py

brian10048 committed:
[IMP] fieldservice: Add team in template

osi-scampbell committed:
[IMP] Add Activites Setting + Template Notebook

[IMP] Remote Group + Alpha Order

fieldservice 12.0.2.3.1

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 99.1% (534 of 539 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

max3903 committed:
[ADD] Spanish (es) translation

fieldservice 12.0.2.3.2

eduaparicio committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 71.2% (384 of 539 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 73.5% (396 of 539 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.0% (512 of 539 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

murtuzasaleh committed:
[12.0][IMP] Improved Color code functionality.

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.0% (512 of 539 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/pt_BR/

fieldservice 12.0.2.3.3

Commit:
Translated using Weblate (Spanish)

Currently translated at 99.1% (536 of 541 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 99.1% (536 of 541 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice/es_CL/

max3903 committed:
[FIX] fieldservice: Keep type when changing template

max3903 committed:
[FIX] fieldservice

max3903 committed:
[IMP] fieldservice: Show locations owned by children (contacts/addresses)

osi-scampbell committed:
[IMP] Location Search Full Name + Parent Loc Dir

[IMP] Flake8

[IMP] Remove Debugging

Update fieldservice/models/fsm_order.py

Co-Authored-By: Maxime Chambreuil <mchambreuil@opensourceintegrators.com>

Update fieldservice/models/fsm_order.py

Co-Authored-By: Maxime Chambreuil <mchambreuil@opensourceintegrators.com>

dreispt committed:
[FIX] FSM Equipment: current location required on View only

Some workflows need to create an FSM Equipemnt that is not yet assigned
to a location.
For example, FSM Equipments still in company warehouse waiting to be
delivered to service locations.

murtuzasaleh committed:
[13.0][MIG][FIX] black/flake8/pylint/isort

murtuzasaleh committed:
[13.0][MIG] fieldservice

brian10048 committed:
[IMP] fieldservice: 'Assigned To' calendar filter

Adds an option in the sidebar of the calendar view to filter orders by the FSM Person that the order is assigned

murtuzasaleh committed:
[IMP] Improved code

max3903 committed:
[IMP] fieldservice: Check holidays

osi-scampbell committed:
[FIX] Report Error

Fixes AttributeError: 'fsm.order' object has no attribute 'state' error when printing an FSO that has an end date.

osi-scampbell committed:
[IMP] Notifications for Stages

max3903 committed:
[FIX] fieldservice: Typo

max3903 committed:
Update res_config_settings.py

marcelsavegnago committed:
[12.0] add internal type in fsm.order.type

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (457 of 457 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

brian10048 committed:
[IMP] fieldservice: Order complete button permission

Issue #588 - FSM User group cannot view Complet button on FSM orders

dreispt committed:
[FIX] fieldservice: fix timezone issue with Order filters

context_today does not work correctly for datetime filters,
it is being used durectly to compare wuth UTC stored datetimes.
To get the expected results, datetime.utcmow() must be used instead.

[FIX] Fix Filters

[IMP] Final Fix

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (467 of 467 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

brian10048 committed:
[IMP] fieldservice order filter names

Fix duplicate filter names

peaseBenjamin committed:
Added a Geo-Locate button to the FSM Location form

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (469 of 469 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

[13.0] [IMP] fieldservice: Minor Changes

Forward port from 12.0

Merge PR #491 into 12.0

Signed-off-by max3903

brian10048 committed:
[IMP] black, prettier

fieldservice 13.0.1.1.0

brian10048 committed:
[IMP] Notification for Stages: adapt to 13

chienandalu committed:
[IMP] fieldservice: basic security

- Add a lower security group whose members only have access to their own
field service orders.
- Add related partner to location and worker forms to easily check the
relationship.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (487 of 487 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (488 of 488 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

brian10048 committed:
[IMP] fieldservice: Minimize partner_fax dependency

Added a separate module to retain the partner_fax feature and eliminated the dependency from the base fieldservice module

fieldservice 13.0.1.2.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (525 of 525 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice/es_AR/

hparfr committed:
[IMP] fieldservice: black, isort, prettier

hparfr committed:
[MIG] fieldservice: Migration to 14.0

hparfr committed:
fieldservice: rename template.hours to template.duration

hparfr committed:
fieldservice: fix typo and fix print menu

HekkiMelody committed:
fixed unlink method in fsm.order

Now evaluates all records in recordset, not just first one

hparfr committed:
fieldservice: remove dead code

hparfr committed:
fieldservice: remove dead code

[UPD] Update fieldservice.pot

[UPD] README.rst

hparfr committed:
fieldservice: add fsm_location as res.partner type

hparfr committed:
fieldservice: remove ref from fsm.location

because ref is already available from res.partner

Commit:
[14.0][FIX] fixed object has no attribute error

Commit:
[14.0][FIX] fsm.person object has no attribute _message_get_suggested_recipients

Commit:
[14.0][FIX] fieldservice: missing underscore

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 41.3% (201 of 486 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/fr_FR/

fieldservice 14.0.1.0.1

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 95.6% (465 of 486 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/fr_FR/

Commit:
[14.0][FIX] Made changes based on review

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 86.2% (453 of 525 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/fr/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 96.7% (508 of 525 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/fr/

[UPD] Update fieldservice.pot

fieldservice 14.0.1.0.2

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

Freni-OSI committed:
[14.0][FIX] fieldservice

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 98.5% (477 of 484 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/fr/

Freni-OSI committed:
[14.0][FIX] fieldservice

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (488 of 488 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

murtuzasaleh committed:
[14.0][FIX] fieldservice

Commit:
Update fieldservice/__manifest__.py

fieldservice 14.0.1.1.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (527 of 527 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (535 of 535 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/pt_BR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (535 of 535 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

tate11 committed:
Added translation using Weblate (Turkish)

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 2.4% (13 of 535 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/tr/

[UPD] Update fieldservice.pot

fieldservice 14.0.1.1.1

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

HekkiMelody committed:
Fixed _compute_request_late methods

[UPD] Update fieldservice.pot

fieldservice 14.0.1.9.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

fieldservice 14.0.1.10.0

fieldservice 14.0.1.11.0

fieldservice 14.0.1.12.0

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 56.5% (303 of 536 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/tr/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (536 of 536 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 60.2% (323 of 536 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/tr/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

Commit:
[14.0][FIX] Edits to res.config.settings for AttributeError

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (553 of 553 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

pedrobaeza committed:
[FIX] fieldservice: Incorrect assignation on onchange

Such field doesn't exists by dependency chain. It should be added on
the child module `fieldservice_stock_request`.

Closes #826

TT31105

hpatelserpentcs committed:
[14.0][FIX]AttributeError: 'res.config.settings' object has no attribute 'group_manage_vendor_price'.

fieldservice 14.0.1.2.0

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

Commit:
Update fieldservice/models/res_config_settings.py

fieldservice 14.0.1.3.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (566 of 566 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

[UPD] Update fieldservice.pot

dreispt committed:
Added translation using Weblate (Portuguese (Portugal))

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

murtuzasaleh committed:
[14.0][FIX] fieldservice

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (568 of 568 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

HekkiMelody committed:
Added float_time widget to duration fields of fsm.order

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 99.8% (567 of 568 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/it/

mourad-ehm committed:
[FIX] by default fsm_category, fsm_tag, fsm_team, fsm_stage are shared by all companies

AmmarOfficewalaSerpentCS committed:
[IMP] Improved a test-cases.

[UPD] README.rst

fieldservice 14.0.1.4.0

HekkiMelody committed:
fsm_order.request_early fix default datetime

HekkiMelody committed:
Removed bad default from order creation

vimalpatelserpentcs committed:
[14.0][FIX]fieldservice: fixup the Field Service Web Timeline View for res config settings

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (568 of 568 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/it/

fieldservice 14.0.1.5.0

brian10048 committed:
[14.0] [IMP] fieldservice: toggle active FSMPerson

[UPD] Update fieldservice.pot

fieldservice 14.0.1.6.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

[UPD] Update fieldservice.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (572 of 572 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

marcos-mendez committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (572 of 572 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/pt_BR/

[UPD] Update fieldservice.pot

fieldservice 14.0.1.7.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

[UPD] Update fieldservice.pot

fieldservice 14.0.1.8.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (573 of 573 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

hparfr committed:
"[IMP] fieldservice: black, isort, prettier

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (574 of 574 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

Freni-OSI committed:
[15.0][FIX] Fixed travis

Freni-OSI committed:
[15.0][MIG] fieldservice

Vandan-OSI committed:
IMP Code for Location.

victoralmau committed:
[IMP] fieldservice: Allow to archive stages.
TT34048

[UPD] Update fieldservice.pot

fieldservice 14.0.1.13.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

victoralmau committed:
[FIX] fieldservice: Apply sudo if user has not access to fsm.location model.
TT34048

fieldservice 14.0.1.13.1

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (575 of 575 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/es_AR/

dreispt committed:
[FIX] fieldservice: fix multicompany

fieldservice 14.0.1.14.0

fieldservice 14.0.1.14.1

hparfr committed:
[FIX] fieldservice fix typo in test

fieldservice 14.0.1.15.0

mymage committed:
[14.0][FIX] fieldservice - worker create

fieldservice 14.0.1.16.0

fieldservice 14.0.1.17.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (575 of 575 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/it/

ygol committed:
fix(fieldservice): adapt access to ir.actions.* records

[IMP] update dotfiles

ygol committed:
fix(fieldservice) address PR remark

hparfr committed:
fieldservice: Fix limit issue

without this fix
some_partner.child_ids.mapped('fsm_location_id')
return just one rec instead of all the expected records

fieldservice 14.0.1.17.1

[UPD] README.rst

fieldservice 14.0.1.17.2

brian10048 committed:
[IMP] fieldservice: Team dashboard

fieldservice 14.0.1.18.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (575 of 575 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/it/

paradoxxxzero committed:
[FIX] fieldservice: Add missing calendar view

In fsm.order action : action_fsm_operation_order

fieldservice 14.0.1.18.1

hparfr committed:
fieldservice: fix Settings link on dashboard

And rename Settings to Configuration to be aligned
with Stock and Account

[UPD] Update fieldservice.pot

fieldservice 14.0.1.19.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

hparfr committed:
fieldservice add person_id in orders view

Murtaza-OSI commented:
> Superseed: #881
> 
> Migration to 15.0
> 
> Please @cesar-tecnativa and @CarlosRoca13 can you review it?
> 
> @Tecnativa TT35713

@victoralmau Any Reason for Super-seed: PR ?

victoralmau commented:
> 

Yes, the contributor has not attended to the changes for 2 months.

Murtaza-OSI commented:
> > 
> 
> Yes, the contributor has not attended to the changes for 2 months.

If there are not then please add log on there PR before you Super-seed PR:
Even I was not unable see any new change after Super-seed in your PR

pedrobaeza commented:
What are you talking about, @Murtaza-OSI ? What is a log? It's clear that if there are comments and they are not attended, the contributor is not maintaining the PR. We have finished the job and respected the attribution, even for the not done changes.

Murtaza-OSI commented:
> What are you talking about, @Murtaza-OSI ? What is a log? It's clear that if there are comments and they are not attended, the contributor is not maintaining the PR. We have finished the job and respected the attribution, even for the not done changes.

I agree with you but you. But ss i see there is nothing been improved in this PR. So is this correct only make new PR without any improvements? And  Super-seed Old PR?

victoralmau commented:
I have separated the changes into 2 commits to make it easier to review.

pedrobaeza commented:
@brian10048 @Murtaza-OSI please review

pedrobaeza commented:
/ocabot migration fieldservice

pedrobaeza commented:
Your commit doesn't have you as author.

victoralmau committed:
[MIG] fieldservice: Migration to 15.0

TT35713

Commit:
[14.0][ADD] fieldservice, fix duplicate/copy method

[UPD] Update fieldservice.pot

fieldservice 14.0.1.19.1

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice/

fieldservice 14.0.1.20.0

victoralmau commented:
Changes done.

pedrobaeza commented:
@Murtaza-OSI @Chanakya-OSI can you review if everything is OK?

brian10048 commented on fieldservice/views/fsm_person.xml:
```diff
@@ -0,0 +1,169 @@
+                    <div name="button_box" class="oe_button_box" />
+                    <label for="name" class="oe_edit_only" />
+                    <h1>
+                        <field name="name" required="True" />
```
```suggestion
                        <field name="name" required="True" />
                        <field name="active" invisible="1" />
```
Enable archive action from the form view action menu

brian10048 commented on fieldservice/views/fsm_person.xml:
```diff
@@ -0,0 +1,169 @@
+                    />
+                </header>
+                <sheet>
+                    <div name="button_box" class="oe_button_box" />
```
```suggestion
                    <div name="button_box" class="oe_button_box" />
                    <widget name="web_ribbon" title="Archived" bg_color="bg-danger" attrs="{'invisible': [('active', '=', True)]}"/>
                    <field name="active_partner" required="0" readonly="1" invisible="1"/>
                        <div class="alert alert-info text-center o_form_header"
                            attrs="{'invisible': [
                                '|', '|',
                                '&amp;', ('active', '=', True), ('active_partner', '=', True),
                                '&amp;', ('active', '=', False), ('active_partner', '=', False),
                                '&amp;', ('active', '=', True), ('active_partner', '=', False),
                            ]}"
                            role="alert">
                          <a class="close" data-dismiss="alert" href="#">x</a>
                          <div>
                            <strong>The contact linked to this user is still active</strong>
                          </div>
                          <div>You can archive the contact
                            <field name="partner_id" required="0" readonly="1"/>
                          </div>
                        </div>
```
Add back the archive ribbon and information

brian10048 reviewed (changes requested):
#881 deleted some basic features and it looks like since that history was kept for this PR that there are still missing features.

brian10048 commented:
@victoralmau Thank you for taking over this migration task.  The improvement for the FSM Stage mixin looks great.  
I made some comments regarding deleted features from the primary migration PR

victoralmau commented:
> @victoralmau Thank you for taking over this migration task. The improvement for the FSM Stage mixin looks great. I made some comments regarding deleted features from the primary migration PR

Thanks, changes done.

brian10048 commented on fieldservice/models/fsm_order.py:
```diff
@@ -0,0 +1,426 @@
+            vals["name"] = self.env["ir.sequence"].next_by_code("fsm.order") or _("New")
+        self._calc_scheduled_dates(vals)
+        if not vals.get("request_late"):
+            if vals.get("priority") == "0":
+                if vals.get("request_early"):
+                    vals["request_late"] = fields.Datetime.from_string(
+                        vals.get("request_early")
+                    ) + timedelta(days=3)
+                else:
+                    vals["request_late"] = datetime.now() + timedelta(days=3)
+            elif vals.get("request_early") and vals.get("priority") == "1":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(days=2)
+            elif vals.get("request_early") and vals.get("priority") == "2":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(days=1)
+            elif vals.get("request_early") and vals.get("priority") == "3":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(hours=8)
```
```suggestion
            vals = self._compute_request_late(vals)
```

victoralmau replied:
The `_compute_request_late()` function is only used for tests, but actually the value of the `request_late` field is defined on creation. I think it would be best to remove it completely, what do you think?

brian10048 commented on fieldservice/views/fsm_order.xml:
```diff
@@ -0,0 +1,491 @@
+                        name="stage_id"
+                        widget="statusbar"
+                        options="{'fold_field': 'fold'}"
+                        domain="[('stage_type', '=', 'order'),
+                                    ('team_ids', 'in', (team_id, False))]"
```
```suggestion
                        domain="[('stage_type', '=', 'order'),
                                    '|', ('team_ids', 'in', team_id), ('team_ids', '=', False)]"
```
Fixes Bug: 
Create a new order stage that has no team (or fsm team group is not enabled)
New stage is in kanban view, but the new stage is not shown in the order form view

brian10048 reviewed (changes requested):
@victoralmau I found a few other minor changes.  Sorry for multiple reviews, but I'm trying to be thorough

victoralmau commented:
Changes done.

victoralmau commented:
Ping @CarlosRoca13

victoralmau commented:
This is ready to merge, right?

hparfr commented on fieldservice/models/fsm_order.py:
```diff
@@ -110,7 +110,7 @@ def _track_subtype(self, init_values):
     location_directions = fields.Char()
     request_early = fields.Datetime(
-        string="Earliest Request Date", default=datetime.now()
+        string="Earliest Request Date", default=fields.Datetime.now
```
are you sure about this ?
It was introduced here https://github.com/OCA/field-service/pull/852

victoralmau replied:
IMO the result would be the same, however, I have left the default as it was before.

hparfr commented:
Hi,


@victoralmau thanks for this work.

You may also include these two PR: 
https://github.com/OCA/field-service/pull/957
https://github.com/OCA/field-service/pull/949

They are ready to be merged, but oca-bot can't merge it for the moment. 
Btw @pedrobaeza you may have the rights to check, it is may be related to the protection of branch v14. 

@victoralmau About the mixin, can you put it in its own commit ? It's hard to distinguish migrations from introduced changes.

victoralmau committed:
[IMP] fieldservice: Add FsmModelMixin to reduce code.

TT35713

victoralmau commented:
> You may also include these two PR: #957 #949

Changes included in commit history.
 
> @victoralmau About the mixin, can you put it in its own commit ? It's hard to distinguish migrations from introduced changes.

I have split the mixin changes into a different commit.

pedrobaeza commented on fieldservice/models/fsm_category.py:
```diff
@@ -16,17 +16,17 @@ class FSMCategory(models.Model):
         "res.company",
         string="Company",
-        required=True,
+        required=False,
```
This change should be reflected in the record rule.

victoralmau replied:
There is really no change from v14 to v15.

An "_unnecessary change_" was done in https://github.com/OCA/field-service/pull/950/commits/090eb57236da3f02a4763157ad41534d9c4f5594, and has been undone in https://github.com/OCA/field-service/pull/950/commits/91eb507729338bf71007c01ad3df6632ff23bced.

Relate to https://github.com/OCA/field-service/pull/881#discussion_r941231783.

pedrobaeza commented on fieldservice/models/fsm_category.py:
```diff
@@ -16,17 +16,17 @@ class FSMCategory(models.Model):
-        required=True,
+        required=False,
         index=True,
-        default=lambda self: self.env.user.company_id,
```
I'm not sure we should make now categories multi-company by default.

victoralmau replied:
https://github.com/OCA/field-service/pull/950#discussion_r988585875

pedrobaeza replied:
OK, but then the default should stay

victoralmau replied:
No, it is something that should not be there because the field is not necessary.

This is how it is in v14 (https://github.com/OCA/field-service/blob/14.0/fieldservice/models/fsm_category.py#L16-L22) and this is how it is in v15 (https://github.com/Tecnativa/field-service/blob/15.0-mig-fieldservice/fieldservice/models/fsm_category.py#L16-L22), there is no difference.

pedrobaeza replied:
Well, not in v13, but OK, at the end is to get used to.

pedrobaeza commented:
/ocabot merge nobump
